### PR TITLE
MileStone3

### PR DESCRIPTION
**Pull request by ioanniskousis**
### - Made the BooksForm component stateful
    - The component stores the title and the book category in its own state
    - The component implements the handleChange method
    - Each change to the form modifys the component’s state
    - The component implements handleSubmit

### - Connected the REMOVE_BOOK action to the BooksList component
    - Added a new table column to the BooksList
    - Implemented the handleRemoveBook method

### CODE REVIEW

File: src/containers/booksForm.js [hunk 1]
```diff
@@ -1,11 +1,80 @@
...
+    });
+  }
+
+  validate(title, category) {
+    const formError = document.getElementById('formError');
+    if (title === '') {
+      formError.innerHTML = "Title can't be empty";
+      setTimeout(() => {
+        formError.innerHTML = '';
+      }, 3000);
```
Comment (suretrust) [inline src/containers/booksForm.js]: **`Optional`** Basing the error message on a timeout is not a good practice, you could use `onBlur` which is more like `offFocus`.

File: src/containers/booksForm.js [hunk 2]
```diff
@@ -1,11 +1,80 @@
...
+      title: '',
+      category: '',
+    });
+  }
 
-export default class BooksForm extends React.Component {
   render() {
     const categoryOptions = [];
     const categoryNames = [
+      '',
```
Comment (suretrust) [inline src/containers/booksForm.js:74]: This should be `'All'`, right?
```suggestion
      'All',
```

File: src/containers/booksForm.js [hunk 3]
```diff
@@ -1,11 +1,80 @@
...
+      title,
+      category,
+    };
+
+    createBook(newBook);
+
+    this.setState({
+      title: '',
+      category: '',
+    });
```
Comment (suretrust) [inline src/containers/booksForm.js:64]: Just for knowledge purpose, you could also use `event.target.reset()`.

File: src/containers/booksList.js [hunk 4]
```diff
@@ -1,21 +1,21 @@
-/* eslint-disable react/forbid-prop-types */
+/* eslint-disable react/prop-types */
```
Comment (suretrust) [inline src/containers/booksList.js]: Add `propTypes` validation, don't disable the rule.

File: src/containers/booksList.js [hunk 5]
```diff
@@ -1,21 +1,21 @@
...
+/* eslint-disable react/prefer-stateless-function */
```
Comment (suretrust) [inline src/containers/booksList.js]: Use a functional component instead, no need to disable the rule.

File: src/containers/booksList.js [hunk 6]
```diff
@@ -1,21 +1,21 @@
...
 
-export default class BooksList extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      books: props.books,
-    };
-  }
-
+class BooksList extends React.Component {
```
Comment (suretrust) [inline src/containers/booksList.js]:
```suggestion
const BooksList = ({ books, removeBook }) => {
```

File: src/components/book.js [hunk 7]
```diff
@@ -1,11 +1,39 @@
...
+import PropTypes from 'prop-types';
+
+export default class Book extends React.Component {
+  render() {
+    const { book, removeBook } = this.props;
+    const { id, title, category } = book;
+
+    const handleRemoveBook = () => {
+      removeBook(id);
+    };
```
Comment (suretrust) [inline src/components/book.js]: This should be implemented in the `BooksList` component. Then you just pass `handleRemoveBook` to the `Book` component as props.

File: src/components/book.js [hunk 8]
```diff
@@ -1,11 +1,39 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+/* eslint-disable react/forbid-prop-types */
+/* eslint-disable react/prefer-stateless-function */
```
Comment (suretrust) [inline src/components/book.js]: Disabling of these rules appear to be unnecessary, let's fix the codebase instead.

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi @ioanniskousis, 👋 

### Great job thus far! 🙌 Please see my comments. 👇 

**KIndly submit another CRR once these issues have been solved.** _Happy coding!_ 🎉

File: src/containers/booksForm.js (hunk 2 above)
Comment (ioanniskousis) [reply src/containers/booksForm.js:74]: thanks but this is for the next milestone4 :)

File: src/containers/booksList.js (hunk 6 above)
Comment (ioanniskousis) [reply src/containers/booksList.js]: DONE

File: src/containers/booksList.js (hunk 5 above)
Comment (ioanniskousis) [reply src/containers/booksList.js]: DONE

File: src/containers/booksList.js (hunk 4 above)
Comment (ioanniskousis) [reply src/containers/booksList.js]: DONE

File: src/containers/booksForm.js (hunk 1 above)
Comment (ioanniskousis) [reply src/containers/booksForm.js]: DONE
used onFocus of fields

File: src/containers/booksForm.js (hunk 3 above)
Comment (ioanniskousis) [reply src/containers/booksForm.js:64]: target is the field not the form.. so reset does not work

File: src/components/book.js (hunk 7 above)
Comment (ioanniskousis) [reply src/components/book.js]: DONE

File: src/components/book.js (hunk 8 above)
Comment (ioanniskousis) [reply src/components/book.js]: DONE

**Review by Torres-ssf — APPROVED** [review]
### Milestone approved :+1: 
Hi @ioanniskousis :hand: 
Great job on this milestone :clap: 
Good luck with the next one :smile: 

Happy Coding :computer: 
**Sergio Torres**

**Pull request merged**